src/components/MultiHopTrade/hooks/useReceiveAddress.tsx

```diff
@@ -11,29 +12,36 @@ import {
   selectBuyAsset,
   selectManualReceiveAddress,
 } from 'state/slices/swappersSlice/selectors'
-import { useAppDispatch, useAppSelector } from 'state/store'
+import { useAppSelector } from 'state/store'
 
-const getReceiveAddress = async ({
-  asset,
-  wallet,
-  accountMetadata,
-}: GetReceiveAddressArgs): Promise<string | undefined> => {
-  const { chainId } = fromAssetId(asset.assetId)
-  const { accountType, bip44Params } = accountMetadata
-  const chainAdapter = getChainAdapterManager().get(chainId)
-  if (!(chainAdapter && wallet)) return
-  const { accountNumber } = bip44Params
-  try {
-    return await chainAdapter.getAddress({ wallet, accountNumber, accountType })
-  } catch (e) {
-    console.log(e)
-  }
-}
+export const getReceiveAddress = pMemoize(
+  async ({
+    asset,
+    wallet,
+    accountMetadata,
+  }: GetReceiveAddressArgs): Promise<string | undefined> => {
+    const { chainId } = fromAssetId(asset.assetId)
+    const { accountType, bip44Params } = accountMetadata
+    const chainAdapter = getChainAdapterManager().get(chainId)
+    if (!(chainAdapter && wallet)) return
+    const { accountNumber } = bip44Params
+    const address = await chainAdapter.getAddress({ wallet, accountNumber, accountType })
+    return address
+  },
+  {
+    cacheKey: ([{ asset, accountMetadata, deviceId }]) => {
+      return JSON.stringify({
```

Comment:
would be best to use stable stringification here, else stringify an array (since js objects are specced to be unordered {even though ackchyually they are stable})

Reply:
f0ddfd35137f84ed46a609082dfcc071d28e5aa4